chain/chunks/src/chunk_cache.rs

```diff
@@ -120,11 +125,33 @@ impl EncodedChunksCache {
         {
             if let Some(chunks_to_remove) = self.height_map.remove(&height) {
                 for chunk_hash in chunks_to_remove {
-                    if !requested_chunks.contains_key(&chunk_hash) {
+                    if !requested_chunks.cache_get(&chunk_hash).is_some() {
                         self.encoded_chunks.remove(&chunk_hash);
                     }
                 }
             }
         }
     }
+
+    pub fn insert_chunk_header(&mut self, shard_id: ShardId, header: ShardChunkHeader) {
+        if header.inner.height_created
```

Comment:
Can the height created be super big? Can I spam with some future heights?

Reply:
Fixed